### ISSUE TITLE
Configure content of MANIFEST.MF and properties file for reproducible builds

### ISSUE THREAD

**Issue by elygre** [nebula-plugins/gradle-info-plugin#43]
We are trying to create a reproducible build, where multiple build creates bit-for-bit identical results, in part to take full advantage of the gradle build cache. However, that means that we need to remove stuff like Build-Date and Build-Number, as well as possibly other stuff.

* Is this possible today?
* Is there any interest in adding this feature to the nebula-info plugins?

Refs:
* https://guides.gradle.org/using-build-cache/#stable_task_inputs
* https://reproducible-builds.org/
* https://discuss.gradle.org/t/how-to-version-control-and-get-reproducible-builds-incl-exactly-the-same-transitive-dependencies/11508

**Comment by rspieldenner** [comment]
Interest yes, I don’t think we have flags to disable them today

**Comment by elygre** [comment]
If I were to get any time free to pursue this, what would be the preferred ways forward?

I think there are two dimensions that need answering:

* Should it be handled as part of collection, or as part of reporting?
* What level of detail do we need? A single "idempotent=true", or more complicated "include <properties> or "exclude <properties>"?

I think the most flexible way is to have an "exclude <properties>"-configuration. Any thoughts?

**Comment by shevek** [comment]
We have a related requirement not for repeatability, but for gradle 4.5.1 build-cache compatibility. Right now, a change to a manifest, e.g. a git commit hash change, causes every project downstream of that JAR to be rebuilt, greatly increasing our build times. We tried adding a normalization rule:
```
      normalization {
              runtimeClasspath {
                      def infoName = 'META-INF/' + project.name + '.properties'
                      ignore infoName;
              }
      }
```
with limited success.

**Comment by elygre** [comment]
I like the solution provided in the merge request.

@DanielThomas , you seem to be the most active contributor. Is this something you could approve and merge?

**Comment by elygre** [comment]
@chali @sghill I see that you two are active committers on this projects. Could you look into this merge request (which solves actual requirements), and perhaps approve it?

**Comment by DanielThomas** [comment]
I'll note that this isn't a problem for the Gradle build cache - the final artifacts aren't, but the class output of the compile tasks.

**Comment by sghill** [comment]
@elygre Released as v3.8.0.
